gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -1636,9 +1650,22 @@ gst_tensor_src_iio_fixate (GstBaseSrc * src, GstCaps * caps)
 {
   /**
    * Caps are fixated based on the device source in _start().
-   * Nothing to be handled here.
    */
-  return gst_caps_fixate (caps);
+  GstTensorSrcIIO *self;
+  self = GST_TENSOR_SRC_IIO_CAST (src);
+  GstCaps *updated_caps = NULL;
+
+  if (gst_caps_can_intersect (caps, self->fixated_caps)) {
+    updated_caps = gst_caps_intersect (caps, self->fixated_caps);
+  } else {
+    GST_ERROR_OBJECT (self,
+        "No intersection between generic and fixated caps of the element.");
+  }
+  gst_caps_unref (caps);
+  gst_caps_unref (self->fixated_caps);
+
+  self->fixated_caps = gst_caps_ref (updated_caps);
```

Comment:
You don't need to ref the returned caps of gst_caps_can_intersect.

Refer: Line 1499 - 1583 of https://github.com/alessandrod/gstreamer/blob/master/gst/gstcaps.c

Reply:
Yes, gst_caps_intersect() increases the refcount before returning.

updated_caps is passed to gst_caps_fixate() (Line 1668), which takes the ownership of the passed caps.
`self->fixated_caps = gst_caps_ref (updated_caps);` intends to store the latest caps for later use (if needed)
That's why I use gst_caps_ref() - if pad unref the caps, self->fixated_caps should still remain valid.

Thanks to your comment, i found 2 bugs 
- caps returned from `gst_caps_fixate (updated_caps)` should be stored, as it is latest and finalized caps
- self->fixated_caps should call 'gst_caps_unref()' in _stop()

Reply:
Moved fixating of caps to _fixate(). This removes unnecessary storage of fixated_caps and calculating fixated caps in _fixate() makes code more readable
This also automatically resolves the two bugs mentioned above.